python/ray/actor.py

```diff
@@ -110,17 +108,60 @@ def annotate_method(method):
     return annotate_method
 
 
+# ActorMethodShell holds static metadata without referencing the handle.
+class ActorMethodShell:
+    def __init__(
+        self,
+        method_name: str,
+        num_returns: Optional[Union[int, Literal["streaming"]]],
+        max_task_retries: int,
+        retry_exceptions: Union[bool, list, tuple],
+        is_generator: bool,
+        generator_backpressure_num_objects: int,
+        enable_task_events: bool,
+        decorator=None,
+        signature: Optional[List[inspect.Parameter]] = None,
+    ):
+        self.method_name = method_name
+        self.num_returns = num_returns
+        self.max_task_retries = max_task_retries
+        self.retry_exceptions = retry_exceptions
+        self.is_generator = is_generator
+        self.generator_backpressure_num_objects = generator_backpressure_num_objects
+        self.enable_task_events = enable_task_events
+        self.decorator = decorator
+        self.signature = signature
+
+    def bind(self, actor_handle):
+        """
+        Produce a bound ActorMethod that holds a strong reference to actor_handle.
+        """
+        return ActorMethod(
```

Comment:
does `ActorMethod` do any significant work in its constructor? we'll want to cache anything that isn't a basic field access

Reply:
there is some logic for setting `num_returns`. I have coped it into the shell class now.